clippy_lints/src/trait_bounds.rs

```diff
@@ -101,3 +146,48 @@ impl<'tcx> LateLintPass<'tcx> for TraitBounds {
         }
     }
 }
+
+fn check_trait_bound_duplication(cx: &LateContext<'_>, gen: &'_ Generics<'_>) {
+    if in_macro(gen.span) {
+        return;
+    }
+
+    let mut map = FxHashMap::default();
```

Comment:
we should early return if:

 - there are no parameters
 - there are no predicates in the where clause